taichi/transforms/whole_kernel_cse.cpp

```diff
@@ -7,12 +7,55 @@ TLANG_NAMESPACE_BEGIN
 
 // Whole Kernel Common Subexpression Elimination
 class WholeKernelCSE : public BasicStmtVisitor {
+ private:
+  std::unordered_set<int> visited;
+  // each scope corresponds to an unordered_set
+  std::vector<std::unordered_set<Stmt *>> visible_stmts;
```

Comment:
Is `std::stack` better than `std::vector` here?

Reply:
`std::stack` doesn't allow access to all elements.